game/hud/src/widgets/Crafting/services/session/job.ts

```diff
@@ -0,0 +1,287 @@
+  type: 'crafting/job/collect',
+  action: () => {
+    return { };
+  },
+  reducer: (s, a) => {
+    // collecting a job, if successful, also clears it
+    const vox = s.vox;
+    return Object.assign(s, initialState(), { vox, status: 'idle' });
+  },
+});
+
+export const setRecipe = module.createAction({
+  type: 'crafting/job/set-recipe',
+  action: (recipe: Recipe) => {
+    return { recipe };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, { recipe: a.recipe });
+  },
+});
+
+export const setQuality = module.createAction({
+  type: 'crafting/job/set-quality',
+  action: (quality: number) => {
+    return { quality };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, { quality: a.quality });
+  },
+});
+
+export const setName = module.createAction({
+  type: 'crafting/job/set-name',
+  action: (name: string) => {
+    return { name };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, { name: a.name });
+  },
+});
+
+export const setMessage = module.createAction({
+  type: 'crafting/job/set-message',
+  action: (message: Message) => {
+    return { message };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, { message: a.message });
+  },
+});
+
+export const setTemplate = module.createAction({
+  type: 'crafting/job/set-template',
+  action: (template: Template) => {
+    return { template };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, { template: a.template });
+  },
+});
+
+export const setPossibleIngredients = module.createAction({
+  type: 'crafting/job/set-possible-ingredients',
+  action: (possible: Ingredient[]) => {
+    return { possible };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, { possibleIngredients: a.possible });
+  },
+});
+
+export function getStatus(callback: (response: any) => void) {
+  if (!isClient()) {
+    callback({
+      status: {
+        vox: '000000003fb7c1f4',
+        type: 'make',
+        status: 'configuring',
+        recipe: null,
+        template: { id: 'item_Arthurian_ArmorMediumForearm01', name: '' },
+        ingredients: [
+          { id: '1', name: 'Sub Iron x20 - 20kg @ 50%', qty: 1 },
+          { id: '2', name: 'Basic Arrow', qty: 1 },
+        ],
+        name: 'La La Land',
+      },
+      complete: 'Running cr on aaac with params [ vox status] ... done.',
+    });
+  } else {
+    slash('cr vox status', (response: any) => {
+      callback(response);
+    });
+  }
+}
+
+export const gotStatus = module.createAction({
+  type: 'crafting/job/got-status',
+  action: (status: VoxStatus) => {
+    return { status };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, {
+      vox: a.status.vox,
+      status: a.status.status,
+      ready: a.status.ready,
+      type: a.status.type,
+      started: a.status.started,
+      endin: a.status.endin,
+      recipe: a.status.recipe,
+      name: a.status.name,
+      template: a.status.template,
+      ingredients: [...a.status.ingredients],
```

Comment:
I guess I am trying to avoid this scenario:-
```
const ingredients = parseIngredients();
const status = { ingredients: ingredients };
dispatch(setStatus(status));
ingredients.pop(); ingredients.shift(); ingredients.splice(0);
```
But maybe I am over thinking it.